### ISSUE TITLE
Typo on RetryMiddlewareFactory

### ISSUE THREAD

**Issue by juanf** [Azure/azure-storage-php#143]
Hi, there's a typo the line linked here,
https://github.com/Azure/azure-storage-php/blob/89e7bcd510c28027c6d8c692c3b90b60d8184531/azure-storage-common/src/Common/Middlewares/RetryMiddlewareFactory.php#L78

should read `Resources::DEFAULT_RETRY_INTERVAL`

Thank you.

Update:
Luckily it's the same on _/azure-storage-common/src/Common/Internal/Resources.php_

**Comment by XiaoningLiu** [comment]
@juanf Aha, thanks for pointing out! Just curious how did you find that? It's a internal constant value, did you try to use it in your project code?

**Comment by juanf** [comment]
Hi, I was just trying to figure out how the RetryMiddleware worked. Do you have any examples or documentation you can refer me to? Thanks!

**Comment by XiaoningLiu** [comment]
@juanf `Middleware` is a concept in Guzzle, an HTTP library we used. You can learn some basic info here http://docs.guzzlephp.org/en/stable/handlers-and-middleware.html

**Comment by juanf** [comment]
Thanks @XiaoningLiu, I was trying to see how it worked specifically on this package, does it retry automatically? do I have to do something like this if I want retries?

            $retry_middleware = RetryMiddlewareFactory::create(
                'General',
                $retries
            );

            $client = BlobRestProxy::createBlobService(
                $connection_string,
                [$retry_middleware]
            );

            $client->createBlockBlob(
                self::$config['container'],
                $file['to'],
                fopen($file['from'], 'r')
            );
Thank you

**Comment by XiaoningLiu** [comment]
@juanf 

```
        $retryMiddleware = RetryMiddlewareFactory::create(
            RetryMiddlewareFactory::GENERAL_RETRY_TYPE,
            3,
            1
        );

        $client = BlobRestProxy::createBlobService(
            $connection_string
        );

        $client->pushMiddleware($retryMiddleware);
```

**Comment by juanf** [comment]
Awesome, thank you very much @XiaoningLiu

**Comment by XiaoningLiu** [comment]
No problem. Will close this issue.

**Comment by XiaoningLiu** [comment]
@juanf This issue has been resolved in latest release azure-storage-common@1.2.0
